### PR TITLE
Add MKLocalSearch to the Location pane

### PR DESCRIPTION
**Pull request by markerpen**
- Replaces the two coordinate textfields in the Location pane with a single search bar.
- The search bar can perform `MKLocalSearch` lookups and center the map on the results.
- The search bar can also detect pasted-in coordinates (`latitude, longitude`) and center the map on those coordinates.
- Added a **Copy** button alongside the coordinates.

https://github.com/twostraws/ControlRoom/assets/9382308/42d5119e-7ee5-4245-8c09-c4baf84fc20e

### CODE REVIEW

File: ControlRoom/Controllers/LocalSearchController.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+     - Parameter for: The partial (autocomplete) query to search for.
+     - Parameter around: Provides a hint for `MKLocalSearchCompleter` to search around a geographical point.
+     - Parameter completion: Called if valid search results are found.
+
+     - Returns: If a location is found immediately (a coordinate was pasted in, for example), returns a `Location`.
+     */
+    func search(for query: String,
+                around location: Location,
+                completion: @escaping ([LocalSearchResult]) -> Void) -> Location? {
```
Comment (Harry-KNIGHT) [inline ControlRoom/Controllers/LocalSearchController.swift:39]:
```suggestion
    func search(
                        for query: String,
                       around location: Location,
                       completion: @escaping ([LocalSearchResult]) -> Void
                       ) -> Location? {
```

Now you have ctrl+M shortcut for make this

File: ControlRoom/Controllers/LocalSearchController.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        guard let callback else { return }
+        let results = completer.results.map {
+            LocalSearchResult( result: $0 )
+        }
+        callback(results)
+    }
+
+    /// Called if `MKLocalSearchCompleter` encounters an error
+    func completer(_ completer: MKLocalSearchCompleter, didFailWithError error: Error) {
+        print(error)
```
Comment (Harry-KNIGHT) [inline ControlRoom/Controllers/LocalSearchController.swift:126]: Why do not throw ?

**Comment by Harry-KNIGHT** [comment]
Really great improvement btw thank you !

File: ControlRoom/Controllers/LocalSearchController.swift (hunk 1 above)
Comment (markerpen) [reply ControlRoom/Controllers/LocalSearchController.swift:39]: Sorry @Harry-KNIGHT I don't see any changes to this code, just some re-formatting. What are you looking to change?

**Comment by twostraws** [comment]
This is fantastic – thank you!

**Pull request merged**